### ISSUE TITLE
ENH clean out the C layer warnings

### ISSUE THREAD

**Issue by beckermr** [LSSTDESC/CCL#672]
Right now we have a lot of C layer warnings which are 

 - not descriptive
 - not correct
 - override other warnings so that the error is not clear at the python layer

We need to clean this up and make sure everything is correct. 

One thing that might be good to do is to use various C macros to denote the file and line number. This will ease the burden of listing the function name in the string. Also, C99 has a standard function name macro that we can use as well.

**Comment by c-d-leonard** [comment]
@beckermr is this resolved by the above merged PR?

**Comment by beckermr** [comment]
Nope. This is still a big issue in the library. The above PR is where we made this issue.

**Comment by c-d-leonard** [comment]
Ah okay thanks, sorry.
